### PR TITLE
refactor(api): use correct sys constraints for OT3

### PR DESCRIPTION
**Pull request by ahiuchingau**
This PR reconfigures the new OT3config by grouping the speed settings (accelerations, max speeds, max speed discontinuity) together. Once we can cache the correct pipette kind based on hardware, the system constraints will be able to load the correctly associated speed settings from the config.

### CODE REVIEW

File: api/src/opentrons/config/types.py [hunk 1]
```diff
@@ -82,16 +85,35 @@ class RobotConfig:
...
 
 
+@dataclass(frozen=True)
+class OT3SpeedSettings:
+    default_max_speed: PerPipetteAxisSettings
+    acceleration: PerPipetteAxisSettings
+    max_speed_discontinuity: PerPipetteAxisSettings
+    direction_change_speed_discontinuity: PerPipetteAxisSettings
+
+    def by_pipette_kind(
```
Comment (Laura-Danielle) [inline api/src/opentrons/config/types.py]: can we change to `gantry_configuration` in this PR?

File: api/src/opentrons/config/defaults_ot3.py [hunk 2]
```diff
@@ -255,24 +256,27 @@ def _build_default_transform(
...
-            robot_settings.get("acceleration", {}), DEFAULT_ACCELERATIONS
-        ),
-        max_speed_discontinuity=_build_default_bpk(
-            robot_settings.get("max_speed_discontinuity", {}),
-            DEFAULT_MAX_SPEED_DISCONTINUITY,
-        ),
-        direction_change_speed_discontinuity=_build_default_bpk(
-            robot_settings.get("direction_change_speed_discontinuity", {}),
-            DEFAULT_DIRECTION_CHANGE_SPEED_DISCONTINUITY,
+        speed_settings=OT3SpeedSettings(
```
Comment (sfoster1) [inline api/src/opentrons/config/defaults_ot3.py]: this is a nit but can we call this motion settings since it also has acceleration

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Can we call it motion settings or something? otherwise looks good!

File: api/src/opentrons/config/types.py (hunk 1 above)
Comment (ahiuchingau) [reply api/src/opentrons/config/types.py]: How about `gantry_load` or something like that? Gantry_configuration.none doesn't quite make sense to me.

Comment (Laura-Danielle) [reply api/src/opentrons/config/types.py]: sure that sounds fine to me

**Review by sfoster1 — APPROVED** [review]
Great!

**Review by Laura-Danielle — APPROVED** [review]
ty

**Pull request merged**